src/mame/snk/hng64_sprite.ipp

```diff
@@ -40,194 +40,424 @@
 
  */
 
-void hng64_state::draw_sprites(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect)
+
+#define PIXEL_OP_REBASE_TRANSPEN(DEST, DESTZ, SOURCE) \
+do \
+{ \
+	if (zval < (DESTZ)) \
+	{ \
+		u32 srcdata = (SOURCE); \
+		if (srcdata != trans_pen) \
+		{ \
+			(DESTZ) = zval; \
+			(DEST) = color + srcdata; \
+		} \
+	} \
+} \
+while (0)
+
+
+#define PIXEL_OP_REBASE_TRANSPEN_REV(DEST, DESTZ, SOURCE) \
+do \
+{ \
+	if (zval > (DESTZ)) \
+	{ \
+		u32 srcdata = (SOURCE); \
+		if (srcdata != trans_pen) \
+		{ \
+			(DESTZ) = zval; \
+			(DEST) = color + srcdata; \
+		} \
+	} \
+} \
+while (0)
+
+#define PIX_CHECKERBOARD \
+if (!checkerboard) \
+{ \
+	srcpix = srcptr[cursrcx >> 16]; \
+} \
+else \
+{ \
+	if (cb & 1) \
+	{ \
+		srcpix = (cury & 1) ? srcptr[cursrcx >> 16] : 0; \
+	} \
+	else \
+	{ \
+		srcpix = (cury & 1) ? 0 : srcptr[cursrcx >> 16]; \
+	} \
+	cb++; \
+}
```

Comment:
Please put the content of this macro inside a `do { ... } while (0)` so that it needs a semicolon after it at point of use and behaves like a single compound statement for tokenising purposes.